src/VisualStudio/LiveShare/Impl/Client/RemoteLanguageServiceWorkspace.cs

```diff
@@ -260,14 +298,19 @@ public Document GetOrAddExternalDocument(string filePath, string language)
             return new DocumentSpan(document, textSpan);
         }
 
-        private Document AddDocumentToProject(string filePath, string language, string projectName)
+        private Document? AddDocumentToProject(string filePath, string language, string projectName)
```

Comment:
Annotation seems odd: would feel like if this function probably should have thrown instead of returning null if something went sideways.